galley/pkg/server/server.go

```diff
@@ -200,7 +206,6 @@ func (s *Server) Wait() error {
 	}
 
 	err := <-s.shutdown
-	s.shutdown = nil
```

Comment:
It is probably better to replace the shutdown channel with a sync.WaitGroup. This should ensure that multiple calls to Wait will be handled appropriately (though Run will not be protected, but that should be fine).